test/unit/transactor/status_test.rb

```diff
@@ -297,5 +297,21 @@ def setup
       assert_not_empty keys
       assert_equal @keys.sort, keys.sort
     end
+
+    test '#to_xml escapes application.redirect_url when OAuth is used' do
+      @application.redirect_url = 'http://example.com?foor=bar&test=true'
+      usage  = {:month => {@metric_id.to_s => 429}}
+      status = Transactor::Status.new(service_id: @service_id,
+                                      application: @application,
+                                      values: usage,
+                                      oauth: true)
+
+      doc = Nokogiri::XML(status.to_xml)
+      redirect_url = doc.at 'redirect_url'
+
+      # If redirect_url is present and the document is still valid is because escaping worked
+      assert_not_nil redirect_url
+      assert_empty doc.errors
```

Comment:
Can we also assert that the XML value parsed to what we have put into it? Just paranoia that document might be valid but parsed into an incorrect value. Although unlikely with the approach used.